_example/servers/idp-server/idp-server.go

```diff
@@ -183,21 +182,19 @@ func main() {
 		},
 	}
 
-	middleware := idp.NewMiddleware(&identityProvider)
-
 	r := chi.NewRouter()
 	r.Use(logHandler)
 
-	r.Get(metadataPath, middleware.ServeMetadata)
+	r.Get(metadataPath, identityProvider.MetadataHandler)
 
 	log.Printf("Test IdP server listening at %s (%s)", *flagListenAddr, *flagPublicURL)
 	switch *flagInitiatedBy {
 	case "idp":
 		r.Get("/", displayLoginForm)
-		r.Get(initiatePath, initiateLogin(middleware))
+		r.Get(initiatePath, initiateLogin(&identityProvider))
 		log.Printf("Go to %s to begin the IdP initiated login.", *flagPublicURL)
 	case "sp":
-		r.Get(ssoPath, middleware.ServeSSO(authFn))
+		r.Get(ssoPath, identityProvider.ServeSSO(authFn))
```

Comment:
@VojtechVitek I guess here we could do something like `r.With(authMiddleware).Get(ssoPath, identityProvider.SignOnHandler)`, right?

Reply:
yes! :)

Reply:
I'll get back to this later as this is not a requirement for the SP case we're handling.